lib/list_versions.go

```diff
@@ -126,17 +109,71 @@ func GetTFURLBody(mirrorURL string) ([]string, error) {
 		os.Exit(1)
 	}
 
-	body, errBody := ioutil.ReadAll(resp.Body)
-	if errBody != nil {
-		log.Printf("[Error] : reading body: %v", errBody)
+	body := new(bytes.Buffer)
+	if _, err := io.Copy(body, resp.Body); err != nil {
+		log.Fatal(err)
+	}
+	if err := json.Unmarshal(body.Bytes(), &releases); err != nil {
+		log.Fatalf("%s: %s", err, body.String())
+	}
+	return releases, nil
+}
+
+func GetTFReleases(mirrorURL string, preRelease bool) ([]Release, error) {
+	// temp testing
+	queryParams := map[string]string{"limit": "20"}
+	rel, _ := getReleases(mirrorURL, queryParams)
+
+	var releases []Release
+	rel, _ = getReleases(mirrorURL, queryParams)
+	releases = append(releases, rel...)
+	for len(rel) == 20 {
+		queryParams["after"] = rel[len(rel)-1].TimestampCreated.String()
+		rel, _ = getReleases(mirrorURL, queryParams)
+		releases = append(releases, rel...)
+	}
+
+	if !preRelease {
+		for i, r := range releases {
+			if r.IsPrerelease {
+				releases = removePreReleases(releases, i)
+			}
+		}
+	}
+	sort.Slice(releases, func(i, j int) bool {
+		return releases[i].Version > releases[j].Version
+	})
+	return releases, nil
+}
+
+func GetTFRelease(mirrorURL, requestedVersion string) (Release, error) {
+	resp, errURL := httpGet(mirrorURL+"/"+requestedVersion, nil)
+	if errURL != nil {
+		log.Printf("[Error] : Getting url: %v", errURL)
+		os.Exit(1)
+		return Release{}, errURL
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != 200 {
```

Comment:
Is this possible to compare with `http`'s `StatusOK` constant instead of relying on assumption that `200` is equal to `OK` within the meaning of `http` package?

Reply:
Agreed, granted this wasn't my code :) but I've updated it everywhere now.